### ISSUE TITLE
assigning specific private ip address to the master node

### ISSUE THREAD

**Issue by porcaro33** [aws/aws-parallelcluster#1302]
**Environment:**
 - AWS ParallelCluster / CfnCluster version [aws-parallelcluster-2.4.1]
 - OS: [CentOS]
 - Scheduler: [SGE]
 - Master instance type: [m5.2xlarge]
 - Compute instance type: [c5.8xlarge]

# this is not bug, enhance request
During setting up a cluster, I do create and delete the cluster several times. Changing master ip is cumbersome for cluster admin(me) and cluster testers(end users). Please allow to assign specific private ip address to the master node in config file.

**Comment by demartinofra** [comment]
Marking this as an enhancement.

Would it be ok for you to assign a given EIP to the master node or you require the private ip to be set to a given one?

**Comment by porcaro33** [comment]
I hope I can set a specific ip to the master node private ip.

**Comment by joeydumont** [comment]
I would also prefer setting a specific private IP.

**Comment by paulojbe** [comment]
Yes. Private IP would be nice. Who opens a HPC cluster to the Internet?

**Comment by hnndlp** [comment]
I would also like to see this feature, it would help a lot during software config. I also don't want a public IP on the cluster.
